tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenWithTransactionTest.cs

```diff
@@ -0,0 +1,156 @@
+            JsonDrivenHelper.EnsureAllFieldsAreValid(document, "name", "object", "arguments", "result");
+
+            base.Arrange(document);
+        }
+
+        // protected methods
+        protected override void AssertResult()
+        {
+            _result.Should().ShouldBeEquivalentTo(_expectedResult.AsBsonDocument);
+        }
+
+        protected override void CallMethod(CancellationToken cancellationToken)
+        {
+            _result = _session.WithTransaction(_callback, _transactionOptions, cancellationToken);
+        }
+
+        protected override async Task CallMethodAsync(CancellationToken cancellationToken)
+        {
+            _result = await _session.WithTransactionAsync(_callbackAsync, _transactionOptions, cancellationToken).ConfigureAwait(false);
+        }
+
+        protected override void SetArgument(string name, BsonValue value)
+        {
+            switch (name)
+            {
+                case "options":
+                    _transactionOptions = ParseTransactionOptions(value.AsBsonDocument);
+                    return;
+                case "callback":
+                    {
+                        var valueDocument = value.AsBsonDocument;
+                        JsonDrivenHelper.EnsureAllFieldsAreValid(valueDocument, "operations");
+                        var operations = valueDocument["operations"].AsBsonArray;
+                        _callback = ParseCallback(operations);
+                        _callbackAsync = ParseCallbackAsync(operations);
+                    }
+                    return;
+            }
+
+            base.SetArgument(name, value);
+        }
+
+        // private methods
+        private TransactionOptions ParseTransactionOptions(BsonDocument document)
+        {
+            JsonDrivenHelper.EnsureAllFieldsAreValid(document, "readConcern", "readPreference", "writeConcern");
+
+            var options = new TransactionOptions();
+            if (document.Contains("readConcern"))
+            {
+                options = options.With(readConcern: ReadConcern.FromBsonDocument(document["readConcern"].AsBsonDocument));
+            }
+
+            if (document.Contains("readPreference"))
+            {
+                options = options.With(readPreference: ReadPreference.FromBsonDocument(document["readPreference"].AsBsonDocument));
+            }
+
+            if (document.Contains("writeConcern"))
+            {
+                options = options.With(writeConcern: WriteConcern.FromBsonDocument(document["writeConcern"].AsBsonDocument));
+            }
+
+            return options;
+        }
+
+        private Func<IClientSessionHandle, CancellationToken, BsonDocument> ParseCallback(BsonArray operations)
+        {
+            return (session, cancellationToken) =>
+            {
+                var tests = ParseCallbackOperations(operations);
+                foreach (var test in tests)
+                {
+                    // TODO: make test aware of session?
```

Comment:
When the Act method calls the underlying operations is it using the correct session? 

The session parameter to the callback is being ignored.

Reply:
Done